flake.nix

```
@@ -0,0 +1,43 @@
+{
+  description = "Lychee - A fast, async, stream-based link checker";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    fenix.url = "github:nix-community/fenix";
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    flake-utils,
+    fenix,
+    ...
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = import nixpkgs {inherit system;};
+      toolchain = fenix.packages.aarch64-darwin.fromToolchainFile {
+        file = ./rust-toolchain.toml;
+        sha256 = "sha256-yMuSb5eQPO/bHv+Bcf/US8LVMbf/G/0MSfiPwBhiPpk=";
+      };
+      platform = pkgs.makeRustPlatform {
+        cargo = toolchain;
+        rustc = toolchain;
+      };
+    in {
+      packages.default = platform.buildRustPackage {
+        pname = "lychee";
+        version = self.rev or self.dirtyShortRev;
+
+        src = pkgs.lib.cleanSource ./.;
+
+        PKG_CONFIG_PATH = "${pkgs.openssl.dev}/lib/pkgconfig";
```

```suggestion
        env.OPENSSL_NO_VENDOR = "1";
```

openssl-sys will pick up on the pkg-config automatically when it's told not to use a vendored build

---

Tried it, got
```
cargo:warning=Could not find directory of OpenSSL installation, and this `-sys` crate cannot proceed without this knowledge. If OpenSSL is installed and this crate had trouble finding it,  you can set the `OPENSSL_DIR` environment variable for the compilation process. See stderr section below for further information.
```
Added `OPENSSL_DIR = pkgs.openssl.dev;` then got 
```
OpenSSL libdir at `["/nix/store/w5z4p9swm9rj7jl955vfnhk4qkkmlmm5-openssl-3.3.2-dev/lib"]` does not contain the required files to either statically or dynamically link OpenSSL
```
Found this [gem](https://discourse.nixos.org/t/rust-pkg-config-issue-on-openssl-dependency/12354/2), works a treat